### PR TITLE
☎️ Implement dial router

### PR DESCRIPTION
**Pull request by rowanseymour**
Adds https://github.com/nyaruka/floweditor/issues/960

Captura de Pantalla 2021-02-01 a la(s) 09 34 39

Captura de Pantalla 2021-02-01 a la(s) 09 34 02

Captura de Pantalla 2021-02-01 a la(s) 09 34 57

### CODE REVIEW

File: src/config/typeConfigs.ts [hunk 1]
```diff
@@ -210,6 +211,14 @@ export const typeConfigList: Type[] = [
     visibility: [FlowTypes.MESSAGING_OFFLINE, FlowTypes.VOICE]
   },
 
+  {
+    type: Types.wait_for_dial,
+    name: i18n.t('actions.wait_for_dial.name', 'Wait for Call'),
+    description: i18n.t('actions.wait_for_dial.description', 'Wait for a call to another number'),
+    form: DialRouterForm,
+    visibility: VISIBILITY_VOICE
```
Comment (rowanseymour) [inline src/config/typeConfigs.ts:219]: @ericnewcomer was thinking to have a component which renders the phone number.. but it seems like components are setup for actions and the other splits don't render anything, so not sure..

File: src/components/flow/routers/dial/DialRouterForm.test.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        $merge: { updateRouter: jest.fn(), onClose: jest.fn() }
+      });
+
+      instance.handlePhoneUpdated('+1234567890');
+      instance.getButtons().secondary.onClick();
+      expect(props.onClose).toHaveBeenCalled();
+      expect(props.updateRouter).not.toHaveBeenCalled();
+    });
+  });
+});
```
Comment (ericnewcomer) [inline src/components/flow/routers/dial/DialRouterForm.test.ts:56]: This is modeled after legacy tests. Instead new code should use `react-testing-library` which doesn't let you interact with the instance instead forcing you to use user events to test. 

Here's an example here:
https://github.com/nyaruka/floweditor/blob/main/src/components/flow/routers/result/ResultRouterForm.test.tsx

File: src/config/typeConfigs.ts (hunk 1 above)
Comment (ericnewcomer) [reply src/config/typeConfigs.ts:222]: As in a temba-component? This seems like a useful thing to include over there (which we could then use directly here) `<temba-phone/>`

**Review by ericnewcomer — APPROVED** [review]
Would like the test to use react-testing-library, but that can also come later as we have a lot of tests still using the current `composeCompontentTests...` approach. I should add a deprecation notice to avoid this confusion.

Are we ready to ship this? Shall I merge it in for today's release?

File: src/config/typeConfigs.ts (hunk 1 above)
Comment (rowanseymour) [reply src/config/typeConfigs.ts:222]: No the component which renders content on the node (e.g. `SendMsgComp`). But we don't do this for other split types (e.g. you can't see the expression on a Split By Expression node) so I think it's ok without it.

**Comment by rowanseymour** [comment]
@ericnewcomer we can't merge this yet because we don't have a way to hide it and @nicpottier is still working on the mailroom side

File: src/config/typeConfigs.ts (hunk 1 above)
Comment (ericnewcomer) [reply src/config/typeConfigs.ts:222]: Ah, well, in any case, seems useful enough to have it live in temba-components so we could use it everywhere.

**Comment by ericnewcomer** [comment]
I suppose we could with an empty visibility flag, but no reason to do it just yet either way

**Review by ericnewcomer — APPROVED** [review]

**Pull request merged**